### PR TITLE
refactor: pool_authority + target

### PR DESCRIPTION
**Pull request by keyvankhademi**

### CODE REVIEW

File: staking/programs/staking/src/context.rs [hunk 1]
```diff
@@ -208,7 +209,7 @@ pub struct CreatePosition<'info> {
         mut,
         seeds = [TARGET_SEED.as_bytes(),&target_with_parameters.get_target().get_seed()[..]],
```
Comment (guibescos) [inline staking/programs/staking/src/context.rs:210]: almost feels like this seed should be hardcoded to be the governance seed

File: staking/tests/config.ts [hunk 2]
```diff
@@ -363,6 +369,58 @@ describe("config", async () => {
         pythTokenListTime: null,
         agreementHash: getDummyAgreementHash(),
         mockClockTime: new BN(30),
+        poolAuthority,
```
Comment (guibescos) [inline staking/tests/config.ts:372]: In this file you should add this to the `someone else tries to access admin methods` as well

**Review by guibescos — APPROVED** [review]
I invite you to modify CreateTarget to only create the governance target

File: staking/programs/staking/src/context.rs (hunk 1 above)
Comment (keyvankhademi) [reply staking/programs/staking/src/context.rs]: Done

File: staking/tests/config.ts (hunk 2 above)
Comment (keyvankhademi) [reply staking/tests/config.ts:370]: Done

**Pull request merged**